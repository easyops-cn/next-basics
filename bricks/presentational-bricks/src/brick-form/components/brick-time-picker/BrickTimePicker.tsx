import React from "react";
import { TimePicker } from "antd";
import { TimePickerProps } from "antd/lib/time-picker";
import { WrapperFormItem } from "../wrapper-form-item/WrapperFormItem";
import { handleChange } from "../../processor";

export interface LegacyBrickTimePickerProps {
  configProps?: TimePickerProps;
  onChange: Function;
}

export function LegacyBrickTimePicker({
  configProps,
  onChange
}: LegacyBrickTimePickerProps): React.ReactElement {
  return (
    <TimePicker {...configProps} onChange={e => handleChange(e, onChange)} />
  );
}
export const BrickTimePicker = WrapperFormItem(LegacyBrickTimePicker);
