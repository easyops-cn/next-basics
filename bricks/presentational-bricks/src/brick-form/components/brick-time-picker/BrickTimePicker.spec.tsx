import React from "react";
import { shallow } from "enzyme";
import { LegacyBrickTimePicker } from "./BrickTimePicker";

describe("BrickTimePicker", () => {
  it("should work", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<LegacyBrickTimePicker onChange={onChange} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should trigger change event", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<LegacyBrickTimePicker onChange={onChange} />);
    wrapper.simulate("change");
    expect(onChange).toHaveBeenCalled();
  });
});
