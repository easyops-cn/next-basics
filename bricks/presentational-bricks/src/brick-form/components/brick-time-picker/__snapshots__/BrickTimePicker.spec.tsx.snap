// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrickTimePicker should work 1`] = `
<TimePicker
  onChange={[Function]}
/>
`;
